odl/discr/diff_ops.py

```diff
@@ -87,7 +87,7 @@ def __init__(self, space, axis=0, method='forward', padding_method=None,
             accuracy in the interior.
         """
         if not isinstance(space, DiscreteLp):
-            raise TypeError('space {!r} is not a DiscreteLp instance.'
+            raise TypeError('`space` {!r} is not a DiscreteLp instance.'
```

Comment:
I dont see why we need to rewrite them in subclasses, users get the full callstack. This is not intended to be the easiest to read error message, it's to allow users to fix their problem in the shortest possible time.

If we use "common language", users may need to reason about which parameter is intended. For example in Chambolle-Pock "step length for the primal variable" or w/e is i.m.o much less clear than `tau` (or `sigma`, even I dont know which one it is).

Another point for this is that it is trivial to refactor if we decide to change the wording or the paramter, just find replace in the file. I found several of this type of error when doing this.

Final comment, if we decide to not capitalize the first letter, we should remove the dots at the end as well, it makes no sense to have dots but not capitalization.